addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CrossDomainScriptInclusionScanRule.java

```diff
@@ -62,7 +70,7 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
                             && isScriptFromOtherDomain(
                                     msg.getRequestHeader().getHostName(), src, msg)) {
                         String integrity = sourceElement.getAttributeValue("integrity");
-                        if (integrity == null || integrity.trim().length() == 0) {
+                        if (integrity == null || integrity.trim().length() == 0 || !trustedDomains.isIncluded(sourceElement.getAttributeValue("href"))) {
```

Comment:
Better if the trust check is done in the previous if statement (note that as is it would raise the alert even if a trusted domain, when the `integrity` was not present or empty).

Reply:
Understood, updated